src/nuclio/projects/project/functions/create-function/function-from-template/function-from-template.component.js

```diff
@@ -149,22 +224,122 @@
                     var msg = 'Oops: Unknown error occurred while getting function\'s templates';
+
                     DialogsService.alert(lodash.get(error, 'data.error', msg));
                 })
                 .finally(function () {
                     ctrl.toggleSplashScreen({value: false});
                 });
         }
+
+        /**
+         * Init data for pagination
+         */
+        function initPagination() {
+            ctrl.page = {
+                number: 0,
+                size: 8
+            };
+
+            paginateTemplates();
+        }
+
+        /**
+         * Gets runtime filters
+         * @returns {Array}
+         */
+        function getRuntimeFilters() {
+            return [
+                {
+                    id: 'all',
+                    name: 'All',
+                    visible: true
+                },
+                {
+                    id: 'golang',
+                    name: 'Go',
+                    visible: true
+                },
+                {
+                    id: 'python:2.7',
+                    name: 'Python 2.7',
+                    visible: true
+                },
+                {
+                    id: 'python:3.6',
+                    name: 'Python 3.6',
+                    visible: true
+                },
+                {
+                    id: 'pypy',
+                    name: 'PyPy',
+                    visible: true
+                },
+                {
+                    id: 'dotnetcore',
+                    name: '.NET Core',
+                    visible: true
+                },
+                {
+                    id: 'java',
+                    name: 'Java',
+                    visible: true
+                },
+                {
+                    id: 'nodejs',
+                    name: 'NodeJS',
+                    visible: true
+                },
+                {
+                    id: 'shell',
+                    name: 'Shell',
+                    visible: true
+                },
+                {
+                    id: 'ruby',
+                    name: 'Ruby',
+                    visible: true
+                }
+            ];
+        }
+
+        /**
+         * Paginates function's templates
+         */
+        function paginateTemplates() {
+
+            // amount of visible items on one page
+            var PAGE_SIZE = 8;
+
+            var filteredTemplates = lodash.chain(templatesOriginalObject)
+                .filter(filterByRuntime)
+                .filter(filterByTitleAndDescription)
+                .value();
+
+            ctrl.templatesWorkingCopy = lodash.chain(filteredTemplates)
+                .slice((ctrl.page.number * PAGE_SIZE), (ctrl.page.number * PAGE_SIZE) + PAGE_SIZE)
+                .keyBy(function (template) {
+                    return template.metadata.name.split(':')[0] + ' (' + template.spec.runtime + ')';
+                })
+                .value();
```

Comment:
Actually, no need to break the chain, you can easily tap into it. You can try this:
```js
ctrl.templatesWorkingCopy = lodash.chain(templatesOriginalObject)
    .filter(filterByRuntime)
    .filter(filterByTitleAndDescription)
    .tap(function (filteredTemplates) {
        ctrl.page.total = Math.ceil(lodash.size(filteredTemplates) / PAGE_SIZE);
    })
    .slice((ctrl.page.number * PAGE_SIZE), (ctrl.page.number * PAGE_SIZE) + PAGE_SIZE)
    .keyBy(function (template) {
        return template.metadata.name.split(':')[0] + ' (' + template.spec.runtime + ')';
    })
    .value();
```

Reply:
For some reason after tap(), slice() won't work properly.
With this code works.
```javascript
            ctrl.templatesWorkingCopy = lodash.chain(templatesOriginalObject)
                .filter(filterByRuntime)
                .filter(filterByTitleAndDescription)
                .thru(function (filteredTemplates) {
                    ctrl.page.total = Math.ceil(lodash.size(filteredTemplates) / PAGE_SIZE);

                    return lodash.slice(filteredTemplates, (ctrl.page.number * PAGE_SIZE), (ctrl.page.number * PAGE_SIZE) + PAGE_SIZE);
                })
                .keyBy(function (template) {
                    return template.metadata.name.split(':')[0] + ' (' + template.spec.runtime + ')';
                })
                .value();
```

Reply:
Okay, good!